src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetLatestDiagnosticsForSpan.cs

```diff
@@ -107,18 +109,137 @@ private class LatestDiagnosticsForSpanGetter
                 }
             }
 
+            private async Task<bool> TryGetSyntaxAndSemanticDiagnosticsAsync(StateSet stateSet, VersionStamp textVersion, VersionStamp syntaxVersion, VersionStamp semanticVersion)
+            {
+                // unfortunately, we need to special case compiler diagnostic analyzer so that 
+                // we can do span based analysis even though we implemented it as semantic model analysis
```

Comment:
@ManishJayaswal @mavasani @JohnHamby @tmeschter I think the root cause of it is that, Diagnostic design team chose least demanding host as the host to design diagnosticanalyzer. since it is the least demanding host, every other most that has much more restrict, demanding requirement, just can't do it probably and do all sort of workaround to meet their environment.

I hope v2 diagnostic analyzer ditch compiler as the base host to design the diagnostic analyzer and choose most depending host as the base host to design the analyzer. and move it down to compiler to support it.